cmake/modules/BuildDPDK.cmake

```diff
@@ -161,21 +166,25 @@ function(do_export_dpdk dpdk_dir)
     list(APPEND DPDK_ARCHIVES "${dpdk_${c}_LIBRARY}")
   endforeach()
 
+  if(NUMA_FOUND)
+    set(dpdk_numa ",-lnuma")
+  endif()
   add_library(dpdk::dpdk INTERFACE IMPORTED)
   add_dependencies(dpdk::dpdk
     ${DPDK_LIBRARIES})
   # workaround for https://gitlab.kitware.com/cmake/cmake/issues/16947
   set_target_properties(dpdk::dpdk PROPERTIES
     INTERFACE_INCLUDE_DIRECTORIES ${DPDK_INCLUDE_DIR}
     INTERFACE_LINK_LIBRARIES
-    "-Wl,--whole-archive $<JOIN:${DPDK_ARCHIVES}, > -Wl,--no-whole-archive")
+    "-Wl,--whole-archive $<JOIN:${DPDK_ARCHIVES}, > -Wl,--no-whole-archive -Wl,-lpthread,-ldl${dpdk_numa}")
```

Comment:
why is `-lpthread,-ldl` needed now?

Reply:
librte_eal.a refers to libpthread.so and libceph-common.so refers to librte_eal.a. Many test programs link libceph-common.so, but without linking pthread, it will cause compilation error. Such as unittest_interval_set, target_link_libraries (unittest_interval_set ceph-common)

Reply:
could you add this in a separated commit?

Reply:
ok,thanks